lib/rules/require-tagless-components.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+const { isEmberComponent } = require('../utils/ember');
+const { isIdentifier, isObjectExpression, isStringLiteral } = require('../utils/types');
+const { isIdentifierImportedFrom } = require('../utils/new-module');
+
+const ERROR_MESSAGE_DEFAULT_TAGNAME = 'Component has a `tagName` of `div` by default';
```

Comment:
The error messages should make clear what the problem is and ideally suggest solutions. I would replace both with this:
```
Please switch to a tagless component by setting `tagName: ''` or converting to a glimmer component
```